project-base/assets/js/frontend/components/lazyLoadInit.js

```diff
@@ -1,17 +1,10 @@
 import MiniLazyload from 'minilazyload';
-import Register from 'framework/common/utils/Register';
 
 /* eslint-disable no-new */
-new MiniLazyload({
+const lazyload = new MiniLazyload({
     rootMargin: '500px',
     threshold: 0.5,
     placeholder: '/assets/frontend/images/noimage.png'
-}, '', MiniLazyload.IGNORE_NATIVE_LAZYLOAD);
+});
 
-export function lazyLoadInit (container) {
-    $(container).find('[loading=lazy]').each(function () {
-        $(this).attr('src', $(this).data('src')).addClass('loaded');
-    });
-}
-
-(new Register()).registerCallback(lazyLoadInit);
+export default lazyload;
```

Comment:
Hmm, why have you decided to go this way instead of registerCallback? I like registerCallback more because it is called on new content automatically. Do you think that it is not necessary to apply lazyload on new contents?

Reply:
RegisterCallback was called after page load and all `src` elements were replaced (so it was not lazy-loaded - every image was loaded after page load). Then it was completely unnecessary to have Minilazyload library. 

Because of the way, how Minilazyload library is implemented I rewrote it to be compatible with the idea of library – new content can be updated via `lazyload.update()` call (source: https://github.com/VelociraptorCZE/MiniLazyload#update) 

by the way, according to the previous documentation, it was still necessary to call `lazyLoadInit($container);` manually on new content.

Reply:
BTW: in previous documentation there were two possible ways to do it - as you say and registerNewContent. 

I am not sure, if this is working right now. because I do not know how export works but only setting that new instance in constant does not initialize that right? Or am I wrong? I see they are describing it like that in docs, but I am not sure.

The part about calling javascripts after page load makes my head spin. When should you actually call that JS? I thought at the end of page is right - so all HTML is loaded and you should apply it then. But you are probably right, that by then there are already some images downloading or downloaded. Also we should solve the situation with new content. It should be automatic and if registerNewContent is not sufficient, we should find some better way.

Reply:
You're right, it was not working for ajax requests without extra `update` call (for example `Load more` button).
Thanks for your help debugging it and how to make it work seamlessly!